kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -170,6 +170,194 @@ static int findMaxColumnId(StructType schema) {
     return maxColumnId;
   }
 
+  /**
+   * The following validation is performed on the updated schema
+   *
+   * <ul>
+   *   <li>Every field has column IDs and physical names, and that these are unique
+   *   <li>Array/map types have valid nested field IDs defined if icebergCompatV2 is enabled
+   *   <li>Physical column names are preserved across the update
+   * </ul>
+   *
+   * @param currentColumnPathToField column path to field for current schema
```

Comment:
can you add a note that this includes all nested/leaf fields please; it wasn't immediately obvious to me :)